mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts

```diff
@@ -15,9 +17,50 @@ export class WayPreviewDAL {
    */
   public static async getWaysPreview(): Promise<WayPreview[]> {
     const waysDTO = await WayService.getWaysDTO();
-    const waysUuids = waysDTO.map((item) => item.uuid);
 
-    const waysPreview = await Promise.all(waysUuids.map(WayPreviewDAL.getWayPreview));
+    const allNeededUsersUuids = new Set(waysDTO.flatMap(wayDTO => [
+      wayDTO.ownerUuid,
+      ...wayDTO.mentorUuids,
+      ...wayDTO.mentorRequestUuids,
+    ]));
+
+    const allNeededGoalsUuids = new Set(waysDTO.flatMap(wayDTO => [wayDTO.goalUuid]));
+
+    const allNeededUsersPreview = await UserPreviewDAL.getUsersPreviewByUuids(Array.from(allNeededUsersUuids));
+
+    const allNeededGoalsPreview = await GoalPreviewDAL.getGoalsPreviewByUuids(Array.from(allNeededGoalsUuids));
+
+    const waysPreview = await Promise.all(waysDTO.map(async (wayDTO) => {
```

Comment:
I dont see async logic inside this Promise.all, why do we need it?

Reply:
Removed